### PR TITLE
Spec-related changes

### PR DESCRIPTION
**Pull request by alexche-cchange**
Add hierarchical, versioned specs. Also clean up some tests.

### CODE REVIEW

File: src/openepd/model/epd.py [hunk 1]
```diff
@@ -30,6 +30,9 @@
 from openepd.model.specs import Specs
 from openepd.model.standard import Standard
 
+OPENEPD_VERSION = "0.1"
+"""Current implemented version of openEPD data format."""
+
 
 class Epd(WithAttachmentsMixin, WithAltIdsMixin, BaseOpenEpdSchema):
```
Comment (corvis) [inline src/openepd/model/epd.py:37]: Maybe we should declare this as `class EpdV0`

And add an alias like `Epd = EpdV0`?

File: src/openepd/model/validation.py [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+#  Unless required by applicable law or agreed to in writing, software
+#  distributed under the License is distributed on an "AS IS" BASIS,
+#  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#  See the License for the specific language governing permissions and
+#  limitations under the License.
+#
+#  This software was developed with support from the Skanska USA,
+#  Charles Pankow Foundation, Microsoft Sustainability Fund, Interface, MKA Foundation, and others.
+#  Find out more at www.BuildingTransparency.org
+#
```
Comment (corvis) [inline src/openepd/model/validation.py:19]: Let's create a `validation` package. We will have a lot of validators for sure and it makes sense to build a structure earlier here.

I would do it like
```
validation
  __init__.py
  numbers.py # <-- RatioFloat goes here
  common.py # <--  together_validator goes here
```

File: src/openepd/model/epd.py (hunk 1 above)
Comment (corvis) [reply src/openepd/model/epd.py:37]: Moreover, we might need to create a factory or factory method/function which allows you to build Epd object from given json/dict.

It should check the version first, and create corresponding class instance.

Comment (alexche-cchange) [reply src/openepd/model/epd.py]: added.

File: src/openepd/model/validation.py (hunk 2 above)
Comment (alexche-cchange) [reply src/openepd/model/validation.py]: done

**Comment by alexche-cchange** [comment]
@corvis I have cherry-picked your commit  and fixed one typing issue. 
Can you pls confirm and if it is good lmk or just merge/release.

**Review by corvis — APPROVED** [review]

**Pull request merged**